language/move-vm/runtime/src/runtime.rs

```diff
@@ -145,21 +142,13 @@ impl VMRuntime {
     }
 }
 
-/// Verify if the transaction arguments match the type signature of the main function.
-fn verify_args(signature: &Signature, args: &[Value]) -> VMResult<()> {
-    if signature.len() != args.len() {
-        return Err(
-            VMStatus::new(StatusCode::TYPE_MISMATCH).with_message(format!(
-                "argument length mismatch: expected {} got {}",
-                signature.len(),
-                args.len()
-            )),
-        );
-    }
-    for (tok, val) in signature.0.iter().zip(args) {
-        if !val.is_valid_script_arg(tok) {
+/// Check that the transaction arguments are acceptable by the VM.
+/// Constants are the only arguments allowed.
+fn check_args(args: &[Value]) -> VMResult<()> {
```

Comment:
at this level we only check that the arguments are restricted.
The interpreter when setting up the initial frame verifies arguments for both scripts and functions